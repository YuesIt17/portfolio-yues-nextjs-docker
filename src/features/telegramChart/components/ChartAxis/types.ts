import {TChartDataLabel} from '@/utils/types';

export type TChartAxis = {
  points: string;
  typeAxis: 'x' | 'y';
  labels: TChartDataLabel[];
  labelPaddingX?: number;
  labelPaddingY?: number;
  maxDataX?: number;
};
