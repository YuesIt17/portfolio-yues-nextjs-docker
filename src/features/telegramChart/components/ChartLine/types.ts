export type TChartLine = {
  points: string;
  fill?: string;
  stroke?: string;
  isVisible?: boolean;
};
