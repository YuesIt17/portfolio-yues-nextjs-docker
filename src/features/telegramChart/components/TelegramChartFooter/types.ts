import {TChartDataLine} from '@/utils/types';

export type TTelegramChartFooter = {
  data: TChartDataLine[] | undefined;
  onChange: (line: TChartDataLine, isVisible: boolean) => void;
};
