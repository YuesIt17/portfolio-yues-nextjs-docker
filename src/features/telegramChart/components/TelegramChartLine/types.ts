import {TChartDataLine, TChartPrepareData} from '@/utils/types';

export type TTelegramChartLine = {
  data: TChartDataLine[];
  width?: number;
  peddingSize?: number;
} & Omit<TChartPrepareData, 'lines'>;
