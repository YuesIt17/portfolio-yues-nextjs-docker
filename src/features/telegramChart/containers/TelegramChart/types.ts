export type TTelegramChart = {
  width: number;
};

export type TTelegramChartStyle = TTelegramChart;
