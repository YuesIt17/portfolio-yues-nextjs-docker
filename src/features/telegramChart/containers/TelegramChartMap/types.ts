import {TChartDataLine} from '@/utils/types';

export type TTelegramChartMap = {
  width: number;
  data: TChartDataLine[];
  maxDataX: number;
  maxDataY: number;
};

export type TTelegramChartMapStyle = {
  width: number;
};
