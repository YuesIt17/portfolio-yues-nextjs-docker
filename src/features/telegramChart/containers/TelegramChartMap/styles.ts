import {css, Theme} from '@emotion/react';
import {TTelegramChartMapStyle} from './types';

export const useStyles = ({width}: TTelegramChartMapStyle) => ({
  chartMap: (theme: Theme) =>
    css({
      width: width,
      height: '100%',
      backgroundColor: theme.colors.secondary.main,
    }),
});
