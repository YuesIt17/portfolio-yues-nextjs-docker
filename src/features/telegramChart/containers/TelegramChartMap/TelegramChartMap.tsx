import React, {useMemo} from 'react';
import {TelegramChartLine} from '../../components';
import {TTelegramChartMap} from './types';
import {CHART_DELTA_MAP_Y} from '../../constants';
import {useStyles} from './styles';
import {TChartDataLine} from '@/utils/types';

const getChartData = (data: TChartDataLine[]) =>
  data.map((item) => {
    return {
      ...item,
      points: item.pointsMap,
    };
  });

export const TelegramChartMap = ({
  data,
  maxDataX,
  maxDataY,
  width,
}: TTelegramChartMap) => {
  const styles = useStyles({width});
  const maxDataMapY = useMemo(
    () =>
      maxDataY >= CHART_DELTA_MAP_Y
        ? Math.round(maxDataY / CHART_DELTA_MAP_Y)
        : 0,
    [maxDataY]
  );
  const chartData = useMemo(() => getChartData(data), [data]);

  if (!chartData) return null;
  return (
    <div css={styles.chartMap} data-testid="telegramChartMap">
      <TelegramChartLine
        data={chartData}
        maxDataX={maxDataX}
        maxDataY={maxDataMapY}
      />
    </div>
  );
};
