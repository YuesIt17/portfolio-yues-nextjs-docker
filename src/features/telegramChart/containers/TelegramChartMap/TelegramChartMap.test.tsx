import React from 'react';
import {themeTestRenderer} from '@/utils/tests';
import {TelegramChartMap} from './TelegramChartMap';
import {TChartDataLine} from '@/utils/types';
import {COLOR_CHART_LINE_JOINED} from '@/share/constants';

const dataInput = [
  {
    color: COLOR_CHART_LINE_JOINED,
    name: 'line_one',
    points: '1,0 6,17',
    pointsMap: '1,0 6,3',
    type: 'line',
    isVisible: true,
  },
] as TChartDataLine[];

describe('TelegramChartMap test', () => {
  test('Check render TelegramChartMap component', () => {
    const telegramChartMap = themeTestRenderer(
      <TelegramChartMap
        data={dataInput}
        maxDataX={10}
        maxDataY={10}
        width={500}
      />
    ).toJSON();

    expect(telegramChartMap).toMatchSnapshot();
  });
});
