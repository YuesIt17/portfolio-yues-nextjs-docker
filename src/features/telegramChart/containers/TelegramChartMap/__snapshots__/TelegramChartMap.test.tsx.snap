// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelegramChartMap test Check render TelegramChartMap component 1`] = `
<div
  className="css-rue32s-chartMap"
  data-testid="telegramChartMap"
>
  <svg
    style={
      Object {
        "width": undefined,
      }
    }
    viewBox="0 0 10 2"
  >
    <polyline
      className="css-bqwtyj-line"
      points="1,0 6,3"
    />
  </svg>
</div>
`;
